kernel/thread.c

```diff
@@ -261,7 +261,6 @@ void z_check_stack_sentinel(void)
 }
 #endif
 
-#ifdef CONFIG_MULTITHREADING
```

Comment:
Is this change related?

Reply:
Yes. Or else the linker would complain about undefined references.

Though, this is just a test to see how to get gcov/codegen to cover those inline functions (hence the draft status). I need to look into the report in details to see if these compiler options actually help or make things worse.